### DISCUSSION TITLE
Action-packed month of ships 🚢 [Community Check-in]

### DISCUSSION THREAD

**Discussion by Akash1134** [community/community#68306] [Actions]
![Screenshot 2023-09-12 at 5 06 08 PM](https://github.com/community/community/assets/95834130/6ed33015-dff8-4a00-a3d2-93a457e031c5)

👋 Ready for another Actions check-in? This week, I’ll share some key additions and changes to Actions, thank you's to our incredible community members, and resources that’ll level up 🆙  your CI/CD automation using Actions.

## 💌  From the Community - Shoutout 📢  to these members

- Great job ✨  by @ChristopherHX for being a top contributor in the Actions space. Though they have contributed to multiple discussions, the one that really stands out where they helped another user get past [how one can run a workflow after another has completed](https://github.com/orgs/community/discussions/66512). 
- Kudos 👍  to @ryanmio for sharing a short and sweet response guiding @petar to the best practices of resolving an [issue with access token with permissions]([https://github.com/orgs/community/discussions/65766](https://github.com/orgs/community/discussions/66930)) Great Job! 💯 
- Let’s look at this self-solve ✔️ - **Much deserving** 👏 applause for @ahmadnassri for updating us with the thorough response and calling out the need to update the [documentation](https://docs.github.com/en/actions/using-workflows/reusing-workflows) about [calling reusable workflows in the same organization](https://github.com/orgs/community/discussions/65766). 

### Transitioning from Node 16 to Node 20

Node 16 has reached its [end of life](https://github.com/nodejs/Release/#end-of-life-releases), prompting us to initiate its deprecation process for GitHub Actions. Our plan is to transition all actions to run on Node 20 by Spring 2024. 

We will actively monitor the migration's progress and gather community feedback before finalizing the transition date. **Starting October 23rd, workflows containing actions running on Node 16 will display a warning to alert users about the upcoming migration.**

#### What you need to do[](#what-you-need-to-do) ❔ 

- For Actions maintainers[](#for-actions-maintainers)
  - Modify your actions to run on Node 20 instead of Node 16\. For guidance, refer to the [Actions configuration settings](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions#runs-for-javascript-actions).

- For Actions users[](#for-actions-users)
  - Ensure your workflows use the latest versions of actions that are running on Node 20\. For more information, see [Using Versions for Actions](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#example-using-versioned-actions).

- For self-hosted runner administrators:[](#for-self-hosted-runner-administrators)
  - Update your self-hosted runners to runner version [v2.308.0](https://github.com/actions/runner/releases/tag/v2.308.0) or later to ensure compatibility with Node 20 actions.

### 📦 Force cancel workflows (Bonus) 

We see a lot of reports in the community where your Actions workflow can become stuck in a state that will not respond to a cancel request. This could block other workflows from executing and would often require customers to contact GitHub Support to resolve the issue.

**Actions customers will now be able to clear stuck workflows by forcing a cancel request from the REST API.** 💥  This is a new feature and the existing endpoint to [cancel a workflow run](https://docs.github.com/rest/actions/workflow-runs#cancel-a-workflow-run) will remain unchanged.

For details, please refer to the [changelog post](https://github.blog/changelog/2023-09-21-github-actions-force-cancel-workflows/). 

### ⬆️  Increased Concurrency Limit for GitHub-Hosted Runners
GitHub-hosted runners now support up to 1000 concurrent jobs for our 4 – 64 vCPU runners, enhancing their capability to handle large-scale CI/CD workloads. 

Enterprise or organization administrators can set this concurrency limit under the [Auto-scaling setting](https://docs.github.com/en/actions/using-github-hosted-runners/managing-larger-runners#configuring-autoscaling-for-larger-runners). GitHub-hosted runners with 4 or more vCPUs are available on both the GitHub Team and Enterprise plans.
Find out more about this in the [changelog post.](https://github.blog/changelog/2023-09-18-increased-concurrency-limit-for-github-hosted-runners/)

### 🆕  [Updated IP ranges for GitHub Enterprise Importer](https://github.blog/changelog/2023-09-08-updated-ip-ranges-for-github-enterprise-importer/)
We're making changes to the IP addresses used by [GitHub Enterprise Importer](https://docs.github.com/en/migrations/using-github-enterprise-importer) for outbound network connections. **These changes went into effect on September 18, 2023.**
- If you're running migrations with GitHub Enterprise Importer and you have IP allowlisting enabled on your migration source or target, or an Azure Blob Storage or Amazon S3 account which you use for migrations, then you'll need to update your allow list.
- Owners of organizations affected by this change were already sent an email notification on August 18, 2023, providing 30 days' notice.

### 🌐  [Dual Static IP ranges for GitHub-hosted Larger runners](https://github.blog/changelog/2023-09-06-dual-static-ip-ranges-for-github-hosted-larger-runners/)
GitHub-hosted larger runners now support dual IP ranges when configured with [Static IPs](https://docs.github.com/en/actions/using-github-hosted-runners/managing-larger-runners#creating-static-ip-addresses-for-larger-runners) for the GitHub Enterprise Cloud plan. 
- With dual IP ranges, larger runner instances will now receive two IP ranges instead of a single range. This enables runners to scale beyond the previously existing 500 concurrency limit. Additionally, the two IP ranges are created in different geographical locations, providing resiliency against regional outages.
- For newly created larger runner instances with the Static IP feature, 2 IP ranges will be assigned by default going forward and no additional action is required. 

🗳️  If you have any feedback to help improve this experience, be sure to post it on this [discussion](https://github.com/orgs/community/discussions/58739).

## Good Reads 📚 - 🔟 Resources to level up your CI/CD automation using Actions

- 🏗️ [How to build a CI/CD pipeline with GitHub Actions in four 4️⃣  simple steps](https://github.blog/2022-02-02-build-ci-cd-pipeline-github-actions-four-steps/)
- 🏃  [Provisioning self-hosted GitHub Actions runners on demand](https://github.com/readme/guides/github-runners-philips)
- 👀 Actions Social Impact - [How GitHub Actions help Ersilia to open source endemic disease research in low-and-middle-income countries](https://socialimpact.github.com/insights/Skills-Based-Volunteering-case-study-with-Ersilia/?utm_source=githubblog&utm_medium=blog&utm_campaign=socialimpact)
- 🚧 [Building organization-wide governance and re-use for CI/CD and automation with GitHub Actions](https://github.blog/2023-04-05-building-organization-wide-governance-and-re-use-for-ci-cd-and-automation-with-github-actions/)
- 🚢 [A curated list of GitHub Actions from community member @sdras](https://github.com/sdras/awesome-actions)
- ♻️ [How to start using reusable workflows with GitHub Actions](https://github.blog/2022-02-10-using-reusable-workflows-github-actions/)
- 4️⃣ [Four tips to keep your GitHub Actions workflows secure](https://github.blog/2023-08-09-four-tips-to-keep-your-github-actions-workflows-secure/)
- 5️⃣  [5 automations every developer should be running](https://github.blog/2021-12-16-5-automations-every-developer-should-be-running/)
- 7️⃣  [7 advanced workflow automation features with GitHub Actions](https://github.blog/2021-11-18-7-advanced-workflow-automation-features-with-github-actions/)
- 🔟  [10 GitHub Actions resources to bookmark from the basics to CI/CD](https://github.blog/2021-11-04-10-github-actions-resources-basics-ci-cd/)

❓  What’s your favorite GitHub Action feature or a cool tip you’d like to share with the community? Post them in the comments below 👇

**Comment by MidhiaQueiroz** [comment]
ok
